climakitae/core/data_export.py

```diff
@@ -67,226 +99,224 @@ def _warn_large_export(file_size, file_size_threshold=5):
-                    dic[k] = ""
-            return dic
 
-        data.attrs = _list_n_none_to_string(data.attrs)
-        for coord in data.coords:
-            data[coord].attrs = _list_n_none_to_string(data[coord].attrs)
-        if "time" in data.coords and "units" in data["time"].attrs:
-            del data["time"].attrs["units"]
+def _fillvalue_encoding(data):
+    """
+    Creates FillValue encoding for each variable for export to NetCDF.
 
-        for data_var in data.data_vars:
-            data[data_var].attrs = _list_n_none_to_string(data[data_var].attrs)
+    Parameters
+    ----------
+    data: xarray.Dataset
 
-    _update_attributes(_data)
+    Returns
+    -------
+    encoding: dict
+    """
+    fill = dict(_FillValue=None)
+    filldict = {coord: fill for coord in data.coords}
+    return filldict
 
-    def _update_encoding(data):
-        """
-        Update data encodings to prevent issues when exporting them to NetCDF.
 
-        Drop `missing_value` encoding, if any, on `data` as well as its coordinates
-        and data variables.
+def _compression_encoding(data):
+    """
+    Creates compression encoding for each variable for export to NetCDF.
 
-        Parameters
-        ----------
-        data: xarray.Dataset
+    Parameters
+    ----------
+    data: xarray.Dataset
 
-        Returns
-        -------
-        None
+    Returns
+    -------
+    encoding: dict
+    """
+    comp = dict(zlib=True, complevel=6)
+    compdict = {var: comp for var in data.data_vars}
+    return compdict
 
-        Notes
-        -----
-        These encoding updates resolve errors raised when writing NetCDF files to
-        S3.
-        """
 
-        def _unencode_missing_value(d):
-            """Drop `missing_value` encoding, if any, on data object `d`.
+def _export_to_netcdf(data, save_name):
+    """
+    Export user-selected data to NetCDF format.
 
-            Parameters
-            ----------
-            d: xarray.Dataset
+    Export the xarray DataArray or Dataset `data` to a NetCDF file `save_name`.
+    If there is enough disk space, the function saves the file locally to the
+    jupyter hub.
 
-            Returns
-            -------
-            None
-            """
-            try:
-                del d.encoding["missing_value"]
-            except:
-                pass
 
-        _unencode_missing_value(data)
-        for coord in data.coords:
-            _unencode_missing_value(data[coord])
+    Parameters
+    ----------
+    data: xarray.DataArray or xarray.Dataset
+        data to export to NetCDF format
+    save_name: string
+        desired output file name, including the file extension
 
-        for data_var in data.data_vars:
-            _unencode_missing_value(data[data_var])
+    Returns
+    -------
+    None
+    """
+    print("Exporting specified data to NetCDF...")
+
+    # Convert xr.DataArray to xr.Dataset so that compression can be utilized
+    _data = data
```

Comment:
Since lines 203 - 218 are almost the same as lines 267 - 281 (aside from a 'NetCDF' vs 'Zarr' string), could we combine both logics into one helper function? Just for conciseness and to make it easier to update this logic in the future

Reply:
Yes, that would be a good thing to adopt. I will work on consolidating the code. Thx

Reply:
I have made a new func `_convert_da_to_ds` to address this in: [4925c91](https://github.com/cal-adapt/climakitae/pull/509/commits/4925c91c4856eac9fe5ac0e3eb978d7e3bb4f057)